homeassistant/components/media_player/bluesound.py

```diff
@@ -0,0 +1,812 @@
+        _add_player(hass, async_add_devices, discovery_info.get('host'),
+                    discovery_info.get('port', None))
+        return
+    else:
+        hosts = config.get(CONF_HOSTS, None)
+        if hosts:
+            for host in hosts:
+                _add_player(hass,
+                            async_add_devices,
+                            host.get(CONF_HOST),
+                            host.get(CONF_PORT, None),
+                            host.get(CONF_NAME, None))
+
+
+class BluesoundPlayer(MediaPlayerDevice):
+    """Bluesound Player Object."""
+
+    def __init__(self, hass, host, port=None, name=None, init_callback=None):
+        """Initialize the media player."""
+        self.host = host
+        self._hass = hass
+        self._port = port
+        self._polling_session = async_get_clientsession(hass)
+        self._polling_task = None  # The actuall polling task.
+        self._name = name
+        self._brand = None
+        self._model = None
+        self._model_name = None
+        self._icon = None
+        self._capture_items = []
+        self._services_items = []
+        self._preset_items = []
+        self._sync_status = {}
+        self._status = None
+        self._last_status_update = None
+        self._is_online = False
+        self._retry_remove = None
+        self._lastvol = None
+        self._init_callback = init_callback
+        if self._port is None:
+            self._port = DEFAULT_PORT
+
+# Internal methods
+    @staticmethod
+    def _try_get_index(string, seach_string):
+        try:
+            return string.index(seach_string)
+        except ValueError:
+            return -1
+
+    @asyncio.coroutine
+    def _internal_update_sync_status(self, on_updated_cb=None,
+                                     raise_timeout=False):
+        resp = None
+        try:
+            resp = yield from self.send_bluesound_command(
+                'SyncStatus',
+                raise_timeout, raise_timeout)
+        except:
+            raise
+
+        if not resp:
+            return None
+        self._sync_status = resp['SyncStatus'].copy()
+
+        if not self._name:
+            self._name = self._sync_status.get('@name', self.host)
+        if not self._brand:
+            self._brand = self._sync_status.get('@brand', self.host)
+        if not self._model:
+            self._model = self._sync_status.get('@model', self.host)
+        if not self._icon:
+            self._icon = self._sync_status.get('@icon', self.host)
+        if not self._model_name:
+            self._model_name = self._sync_status.get('@modelName', self.host)
+
+        if on_updated_cb:
+            on_updated_cb()
+        return True
+# END Internal methods
+
+# Poll functionality
+    @asyncio.coroutine
+    def _start_poll_command(self):
+        """"Loop which polls the status of the player."""
+        try:
+            while True:
+                yield from self.async_update_status()
```

Comment:
I don't want to set the same scan interval for every request. There is no need to make every request every time. I could achieve this with timestamps, but I thought this was a better and more easy to read.

My thought with this was to use the poll request (which I accidentally turned off, I will change this) as timer together with the throttle attribute